app/src/full/kotlin/io/homeassistant/companion/android/sensors/LocationSensorManager.kt

```diff
@@ -928,18 +928,24 @@ class LocationSensorManager :
         if (updateLocationAs == SEND_LOCATION_AS_ZONE_ONLY) {
             val zones = getZones(serverId)
             val locationZone = zones
-                .filter { !(it.attributes["passive"] as Boolean) && it.containsWithAccuracy(location) }
-                .minByOrNull { (it.attributes["radius"] as Number).toFloat() }
+                .filter {
+                    val passive = it.attributes["passive"] as? Boolean ?: true
+                    val radius = it.attributes["radius"] as? Number
+                    return@filter !passive && it.containsWithAccuracy(location) && radius != null
```

Comment:
... and then explicitly filter for where the value is `false`, no assumption about the default/fallback needed.

```suggestion
                    return@filter passive == false && it.containsWithAccuracy(location) && radius != null
```